specs/dotsigstore_bundle/pb/sigstore_bundle.proto

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.bundle.v1;
+
+// https://raw.githubusercontent.com/secure-systems-lab/dsse/9c813476bd36de70a5738c72e784f123ecea16af/envelope.proto
+import "envelope.proto";
+import "sigstore_common.proto";
+import "sigstore_rekor.proto";
+
+option go_package = "github.com/sigstore/proto/bundle/v1";
+option java_package = "dev.sigstore.proto.bundle.v1";
+option java_multiple_files = true;
+option java_outer_classname = "BundleProto";
+
+// Notes on versioning.
+// The primary message ('Bundle') MUST be versioned, by populating the
+// 'media_type' field. Semver-ish (only major/minor versions) scheme MUST
+// be used. The current version as specified by this file is:
+// application/vnd.dev.sigstore.bundle+json;version=0.1
+// The semantic version is thus '0.1'.
+
+// This message holds different constructs that captures the timestamp
+// of the time the signature was generated.
+// As this message can be either empty (no timestamps), or a combination of
+// an arbitrarily number of transparency log entries and signed timestamps,
+// it is the client's responsibility to implement any required verification
+// policies.
+// When using RFC3161 timestamps together with DSSE envelopes, the message
+// imprint (message digest) MUST be computed over the payload (octets).
+message TimestampVerificationData {
+        // This is the regular inclusion promise and proof, where
+        // the timestamp is coming from the transparency log.
+        repeated dev.sigstore.rekor.v1.TransparencyLogEntry tlog_entries = 1;
+        // A list of RFC3161 signed timestamps provided by the user.
+        // This can be used when the entry has not been stored on a
+        // transparency log.
+        // Clients MUST verify the hashed message in the message imprint
+        // against the artifact. Note that the message hash algorithm may
```

Comment:
Yes, this is something I have though a bit over too and I think I was too quick here. I think having the TSA witness the signature bytes is the most convenient and secure method, as it would not depend on any possible canonicalization of the artifact itself (this can still be an issue for the signature computation though, but it's not relevant for this).

Reply:
IMO it makes the most sense to have the TSA witness the Rekor entry, for symmetry with the TransparencyLogEntry.

Reply:
But there is no requirement that a Rekor entry exists, so I would argue that a RFC3161 timestamp can be used instead (or complementary too of course) of a Rekor entry to get a verified timestamp.

Reply:
Yes, I agree, we shouldn't link the 3161 signature to Rekor. Following other code signing specs, I would recommend that it's over the signature of the artifact.